tfhe/src/integer/oprf.rs

```diff
@@ -0,0 +1,76 @@
+use super::{RadixCiphertext, ServerKey};
+use crate::core_crypto::commons::generators::DeterministicSeeder;
+use crate::core_crypto::prelude::ActivatedRandomGenerator;
+use rayon::iter::{IndexedParallelIterator, IntoParallelIterator, ParallelIterator};
+
+pub use concrete_csprng::seeders::{Seed, Seeder};
+
+impl ServerKey {
+    /// Generates a `num_block` blocks integer
+    /// taken uniformly in `[0, 2^random_bits_count[` using the given a seed
+    /// It can be useful to make server random generation deterministic
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    /// use tfhe::Seed;
+    ///
+    /// let size = 4;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size);
+    ///
+    /// let random_bits_count = 3;
+    ///
+    /// let msg = sks.par_generate_oblivious_pseudo_random(Seed(0), random_bits_count, size as u64);
+    ///
+    /// // Decrypt:
+    /// let dec_result = cks.decrypt(&ct_res);
+    /// assert!(dec_result < (1 << random_bits_count));
+    /// ```
+    pub fn par_generate_oblivious_pseudo_random(
```

Comment:
Added `_unsigned_integer` in the end of the name
I'm not sure how to handle the signed case

Reply:
I think you can use the from_blocks on SignedRadixInteger, so basically the same function but wrapping differently, for the data range it will be much more tricky, so I guess we could start with the full range for the signed integer as a baseline, I think there would need to be a sign bit to manage, maybe worth a discussion with Thomas